api/src/main/java/org/apache/iceberg/expressions/UnboundTransform.java

```diff
@@ -26,7 +26,7 @@ public class UnboundTransform<S, T> implements UnboundTerm<T>, Term {
   private final NamedReference<S> ref;
   private final Transform<S, T> transform;
 
-  UnboundTransform(NamedReference<S> ref, Transform<S, T> transform) {
+  public UnboundTransform(NamedReference<S> ref, Transform<S, T> transform) {
```

Comment:
You can use `Expressions .transform`

Reply:
Ack, I have addressed this here: https://github.com/apache/iceberg/pull/13032/files#diff-7db539f11ce6db77f67fc6e6e414c39a42891eefc1eb88588035532ff31704baR124